core/tests/protractor_desktop/userJourneys.js

```diff
@@ -265,14 +265,14 @@ describe('Site language', function() {
       await browser.get('/collection/' + collectionId);
       await waitFor.pageToFullyLoad();
       expect(await element(by.css('.oppia-share-collection-footer')).getText())
-        .toEqual('SHARE THIS COLLECTION');
+        .toEqual('COMPARTIR ESTA COLECCIÓN');
```

Comment:
Are you sure this is correct? The test title says `should not change in exploration and collection player for guest users`, so we should not expect the language to change to Spanish here

Reply:
Done already

Reply:
This is the current behaviour. Is the present behaviour wrong, should I fix this?

Reply:
Before my changes, we were expecting languages to be spanish. As the test title and implementation were contradicting each other. so, I changed it to english but it caused e2e to fail. So, i revert to spanish back again and left a comment on the PR which git blame showed me to verify that the current behaviour is right or not? But I still dont have a proper reply. I am not sure whether the e2e test title is wrong or current translation behaviour for exploration page is wrong. Please verify.

Reply:
Sorry, I was looking at only the recent commits. I don't know what the expected behavior is here. @aks681 @kevintab95 when a user is not logged in, do we expect language changes to affect the exploration and collection player pages? Our E2E tests are currently named as if we do not expect this change, but the E2E code is written as if we do expect this change

Reply:
I think what the test means is, if you select language X and then view the exploration or collection viewer, the language X should remain selected (i.e. it shouldn't change back to the previous selection).

Reply:
OK, that makes sense. I'll approve to unblock this PR